apps/server/src/shared/domain/entity/task.entity.ts

```diff
@@ -99,102 +104,153 @@ export class Task extends BaseEntityWithTimestamps implements ILearnroomElement,
 		this.publicSubmissions = props.publicSubmissions || false;
 	}
 
-	isFinishedForUser(user: User): boolean {
-		return !!(this.finished?.contains(user) || this.course?.isFinished());
+	private getSubmissionItems(): Submission[] {
+		if (!this.submissions || !this.submissions.isInitialized(true)) {
+			throw new InternalServerErrorException('Submissions items are not loaded.');
+		}
+		const submissions = this.submissions.getItems();
+
+		return submissions;
+	}
+
+	private getFinishedUserIds(): EntityId[] {
+		const finishedObjectIds = this.finished.getIdentifiers('_id');
+		const finishedIds = finishedObjectIds.map((id): string => id.toString());
+
+		return finishedIds;
 	}
 
-	isDraft(): boolean {
+	private getParent(): ITaskParent | User {
```

Comment:
I had it first if you want to look to the initial commits from me. 
It bring us into the situation that the user must be implement getStudentIds: EntityId[]  and the user do not know his one role in this situations. 
If you look to the answer from me to @dyedwiper . I am extrem unhappy to implement getStudentUserIds instead of getNumberOfStudents().

I really like if we implement the interfaces also on user, yes maybe we must be add 20+ interfaces in the end to the user. But it is how our datastructure work and the code should be show it. But for user is a teacher the comment user.getStudentUserIds() return then a [user.id] ...that is also not correct. 

I implement the compromise with if else check for task.getMaxSubmissions. I know that is bring us into the situation that we must implement on any method that something want from the parent also must be implement a if else. But we must look into it later. It is a bigger topic and hard to solve in a small ticket that only move code.